routes/account.js

```diff
@@ -80,6 +83,86 @@ router.post('/logout', function(req, res) {
     res.sendStatus(200);
 });
 
+router.get('/updateProfile', verifyJWTCookie, function(req, res) {
+    UserModel.findOne({id: req.user.id}, function(err, user) {
+        if (err) {
+            return res.status(422).json( {success: false});
+        }
+        if (user) {
+            res.render('updateProfile', {username: user.username, 
+                                           firstname: user.firstname, 
+                                           lastname: user.lastname,
+                                           email: user.email, 
+                                           password: user.password})
+            
+        } else {
+            res.status(401).json( {success: false, invalidCredentials: true});
+        }
+    });
+});
+
+router.post('/updateProfile', verifyJWTCookie, function(req, res) {
+    var updateUser = new UserModel();
+    
+    UserModel.findOne({id: req.user.id}, function(err, user) {
+        if (err) {
+            return res.status(422).json( {success: false});
+        }
+        
+        if (user) {
+            updateUser = user;
+
+            //if a value exists in the updateProfile_form then change the users 
+            //information to match it
+            if (req.body.username) {
+                updateUser.username = req.body.username;
+            }
+            if (req.body.firstname) {
+                updateUser.firstname = req.body.firstname;
+            }
+            if (req.body.lastname) {
+                updateUser.lastname = req.body.lastname;
+            }
+            if (req.body.email) {
+                updateUser.email = req.body.email;
+            }
+            //maybe works, maybe not
+            if (req.body.password) {
+                updateUser.setPassword (req.body.password);
+            }
+           
+            UserModel.findOneAndUpdate({id: req.user.id}, {
```

Comment:
You can just call `user.save` here instead. You do not need to find the record, because you are in the callback of `findOne` already.